File: ganttproject-tester/test/net/sourceforge/ganttproject/TestSetupHelper.java
package net.sourceforge.ganttproject;

import biz.ganttproject.core.calendar.AlwaysWorkingTimeCalendarImpl;
import biz.ganttproject.core.calendar.GPCalendarCalc;
import biz.ganttproject.core.option.ColorOption;
import biz.ganttproject.core.option.DefaultColorOption;
import biz.ganttproject.core.time.CalendarFactory;
import biz.ganttproject.core.time.GanttCalendar;
import biz.ganttproject.core.time.TimeUnitStack;
import biz.ganttproject.core.time.impl.GPTimeUnitStack;
import net.sourceforge.ganttproject.gui.NotificationManager;
import net.sourceforge.ganttproject.resource.HumanResourceManager;
import net.sourceforge.ganttproject.roles.RoleManager;
import net.sourceforge.ganttproject.roles.RoleManagerImpl;
import net.sourceforge.ganttproject.task.CustomColumnsManager;
import net.sourceforge.ganttproject.task.TaskManager;
import net.sourceforge.ganttproject.task.TaskManagerConfig;

import java.awt.*;
import java.net.URL;

public class TestSetupHelper {
    public static class TaskManagerBuilder implements TaskManagerConfig {
        private GPCalendarCalc myGPCalendar = new AlwaysWorkingTimeCalendarImpl();

        private TimeUnitStack myTimeUnitStack;

        private HumanResourceManager myResourceManager;

        private RoleManager myRoleManager;

        private DefaultColorOption myDefaultColorOption = new DefaultColorOption("taskcolor", Color.CYAN);

        public TaskManagerBuilder() {
            myTimeUnitStack = new GPTimeUnitStack();
            myRoleManager = new RoleManagerImpl();
            myResourceManager = new HumanResourceManager(myRoleManager.getDefaultRole(), new CustomColumnsManager(), myRoleManager);
        }

        @Override
        public Color getDefaultColor() {
            return myDefaultColorOption.getValue();
        }

        @Override
        public ColorOption getDefaultColorOption() {
        return myDefaultColorOption;
      }

        @Override
        public GPCalendarCalc getCalendar() {
            return myGPCalendar;
        }

        @Override
        public TimeUnitStack getTimeUnitStack() {
            return myTimeUnitStack;
        }

        @Override
        public HumanResourceManager getResourceManager() {
            return myResourceManager;
        }

        @Override
        public URL getProjectDocumentURL() {
            return null;
        }

        public TaskManagerBuilder withCalendar(GPCalendarCalc calendar) {
            myGPCalendar = calendar;
            return this;
        }

        public TaskManager build() {
            return TaskManager.Access.newInstance(null, this);
        }

        @Override
        public NotificationManager getNotificationManager() {
          return null;
        }
    }

    public static TaskManagerBuilder newTaskManagerBuilder() {
        return new TaskManagerBuilder();
    }

    public static GanttCalendar newFriday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 15);
    }

    public static GanttCalendar newSaturday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 16);
    }

    public static GanttCalendar newSunday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 17);
    }

    public static GanttCalendar newTuesday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 19);
    }

    public static GanttCalendar newMonday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 18);
    }

    public static GanttCalendar newWendesday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 20);
    }

    public static GanttCalendar newThursday() {
        return CalendarFactory.createGanttCalendar(2004, 9, 21);
    }


}

File: cloud.ganttproject.colloboque/src/test/kotlin/cloud/ganttproject/colloboque/TestSetupHelper.kt
/*
Copyright 2022 BarD Software s.r.o, Edgar Zhavoronkov

This file is part of GanttProject, an opensource project management tool.

GanttProject is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.

GanttProject is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
*/
package cloud.ganttproject.colloboque;

import biz.ganttproject.core.calendar.AlwaysWorkingTimeCalendarImpl
import biz.ganttproject.core.calendar.GPCalendarCalc
import biz.ganttproject.core.option.BooleanOption
import biz.ganttproject.core.option.ColorOption
import biz.ganttproject.core.option.DefaultBooleanOption
import biz.ganttproject.core.option.DefaultColorOption
import biz.ganttproject.core.time.TimeUnitStack
import biz.ganttproject.core.time.impl.GPTimeUnitStack
import net.sourceforge.ganttproject.gui.NotificationManager
import net.sourceforge.ganttproject.resource.HumanResourceManager
import net.sourceforge.ganttproject.roles.RoleManager
import net.sourceforge.ganttproject.roles.RoleManagerImpl
import net.sourceforge.ganttproject.storage.ProjectDatabase
import net.sourceforge.ganttproject.task.CustomColumnsManager
import net.sourceforge.ganttproject.task.TaskManager
import net.sourceforge.ganttproject.task.TaskManagerConfig
import java.awt.Color
import java.net.URL


object TestSetupHelper {
  fun newTaskManagerBuilder(): TaskManagerBuilder {
    return TaskManagerBuilder()
  }

  class TaskManagerBuilder : TaskManagerConfig {
    private var myGPCalendar: GPCalendarCalc = AlwaysWorkingTimeCalendarImpl()
    private val myTimeUnitStack: TimeUnitStack
    private val myResourceManager: HumanResourceManager
    private val myRoleManager: RoleManager
    private val myDefaultColorOption = DefaultColorOption("taskcolor", Color.CYAN)
    private val mySchedulerDisabledOption = DefaultBooleanOption("scheduler.disabled", false)
    private var taskUpdateBuilderFactory: ProjectDatabase.TaskUpdateBuilder.Factory? = null

    init {
      myTimeUnitStack = GPTimeUnitStack()
      myRoleManager = RoleManagerImpl()
      myResourceManager = HumanResourceManager(myRoleManager.defaultRole, CustomColumnsManager(), myRoleManager)
    }

    override fun getDefaultColor(): Color {
      return myDefaultColorOption.value!!
    }

    override fun getDefaultColorOption(): ColorOption {
      return myDefaultColorOption
    }

    override fun getCalendar(): GPCalendarCalc {
      return myGPCalendar
    }

    override fun getTimeUnitStack(): TimeUnitStack {
      return myTimeUnitStack
    }

    override fun getResourceManager(): HumanResourceManager {
      return myResourceManager
    }

    override fun getProjectDocumentURL(): URL? {
      return null
    }

    fun build(): TaskManager {
      return TaskManager.Access.newInstance(null, this, taskUpdateBuilderFactory)
    }

    override fun getNotificationManager(): NotificationManager? {
      return null
    }

    override fun getSchedulerDisabledOption(): BooleanOption {
      return mySchedulerDisabledOption
    }

    fun setTaskUpdateBuilderFactory(factory: ProjectDatabase.TaskUpdateBuilder.Factory?) {
      taskUpdateBuilderFactory = factory
    }
  }
}
